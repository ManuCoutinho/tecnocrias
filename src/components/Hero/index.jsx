import { useEffect, useRef, useState } from 'react'
import RINGS from 'vanta/dist/vanta.rings.min.js'
import styles from './styles.module.scss'

function HeroAnimation(){
  const [vantaEffect, setVantaEffect] = useState(null)
  const vantaRef = useRef(null)

  useEffect(() => {
    if(!vantaEffect){
      setVantaEffect(() =>
        RINGS({
          el: vantaRef.current,
          mouseControls: true,
          touchControls: true,
          gyroControls: false,
          minHeight: 200.0,
          minWidth: 200.0,
          scale: 1.0,
          scaleMobile: 1.0,
          backgroundColor: 0x16161a,
          color: 0x2cb67d
        })
      )
    }

    return () => {
      if(vantaEffect) vantaEffect.destroy()
    }
  }, [vantaEffect])

  return (
    <section ref={vantaRef} className={styles.hero__container}>
      <div>
        <p>
          VocÃª Ã© apaixonade por tecnologia e estÃ¡ ansiose para expandir seus
          conhecimentos no mundo do cÃ³digo aberto? NÃ£o procure mais!
        </p>
        <p>
          O Tecnocrias estÃ¡ te convocando para embarcar em uma jornada incrÃ­vel.
          ğŸš€
        </p>
      </div>
    </section>
  )
}

export default HeroAnimation
