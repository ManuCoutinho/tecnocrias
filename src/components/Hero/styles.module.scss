@import '../../styles/base/variables';
@import '../../styles/mixins/animations';

.hero__container {
  min-width: 100%;
  max-width: 90dvw;
  height: 95dvh;
  width: 100%;
  position: relative;

  & > div {
      font-size: $lg;
      padding: 1rem;
      padding-left: 10rem;
      word-wrap: break-word;
      white-space: pre-line;
      color: $secondary;
      max-width: 52rem;
      line-height: 2.5rem;
      font-weight: $semi_bold;

      @include slideInLeft();
      & p {
            margin-top: 50%;
      }
    }

}
