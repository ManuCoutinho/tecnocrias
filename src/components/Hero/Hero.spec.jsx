import React from 'react'
import '@testing-library/jest-dom'
import { screen, render } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'
import Hero from '.'

vi.spyOn(React, 'useEffect')
vi.spyOn(React, 'useState')
vi.spyOn(React, 'useRef')
vi.mock('vanta/dist/vanta.rings.min.js')
describe('<Hero/>', () => {
  it('should render Hero correctly', () => {
    render(<Hero/>)
    const description = screen.getByText(
      /O Tecnocrias está te convocando para embarcar em uma jornada incrível/i
    )
    const container = screen.getByTestId('hero')
    expect(container).toBeInTheDocument()
    expect(description).toBeInTheDocument()
    expect(container).toHaveClass('hero__container')
  })
  it('should render Hero correctly', () => {
    const {container} = render(<Hero />)
    expect(container).toMatchSnapshot()

  })
 })
