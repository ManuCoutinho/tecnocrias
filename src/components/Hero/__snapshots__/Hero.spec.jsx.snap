// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Hero/> > should render Hero correctly 1`] = `
<div>
  <section
    class="hero__container"
    data-testid="hero"
  >
    <div>
      <p>
        VocÃª Ã© apaixonade por tecnologia e estÃ¡ ansiose para expandir seus conhecimentos no mundo do cÃ³digo aberto? NÃ£o procure mais! 
        <br />
        O Tecnocrias estÃ¡ te convocando para embarcar em uma jornada incrÃ­vel.ğŸš€
      </p>
    </div>
  </section>
</div>
`;
